src/helm/proxy/clients/openai_client.py

```diff
@@ -71,7 +73,32 @@ def make_request(self, request: Request) -> RequestResult:
                 # to be returned in a single assistant message.
                 # TODO: Support ChatML for creating multiple messages with different roles.
                 # See: https://github.com/openai/openai-python/blob/main/chatml.md
-                messages = [{"role": "user", "content": request.prompt}]
+
+                # Content can either be text or a list of multimodal content made up of text and images:
+                # https://platform.openai.com/docs/guides/vision
+                content: Union[str, List[Union[str, Any]]]
+                if request.multimodal_prompt is not None:
+                    content = []
+                    for media_object in request.multimodal_prompt.media_objects:
+                        if media_object.is_type("image") and media_object.location:
+                            from helm.common.images_utils import encode_base64
+
+                            base64_image: str = encode_base64(media_object.location)
+                            content.append(
+                                {"type": "image_url", "image_url": {"url": f"data:image/jpeg;base64,{base64_image}"}}
+                            )
+                        elif media_object.is_type(TEXT_TYPE):
+                            if media_object.text is None:
+                                raise ValueError("MediaObject of text type has missing text field value")
+                            content.append({"type": media_object.type, "text": media_object.text})
+                        else:
+                            raise ValueError(f"Unrecognized MediaObject type {media_object.type}")
+
+                else:
+                    content = request.prompt
+
+                messages = [{"role": "user", "content": content}]
+
             raw_request = {
                 "model": request.model_engine,
                 "messages": messages,
```

Comment:
So here, compared to other clients we are using the raw messages with the base 64 images as cache keys instead of using the `generate_uid` function used by other VLM clients. Is there a reason to not do like the others? Just asking to make sure I understand

Reply:
I updated OpenAI's client for consistency.